application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,209 @@
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.apache.commons.collections4.ListUtils;
+import org.kohsuke.github.*;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.awt.*;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * GitHub Referencing feature. If someone sends #id of an issue (e.g. #207) in specified channel,
+ * the bot replies with an embed that contains info on the issue/PR.
+ */
+public final class GitHubReference extends MessageReceiverAdapter {
+    static final String ID_GROUP = "id";
+
+    /**
+     * The pattern(#123) used to determine whether a message is referencing an issue.
+     */
+    static final Pattern ISSUE_REFERENCE_PATTERN =
+            Pattern.compile("#(?<%s>\\d+)".formatted(ID_GROUP));
+    private static final int ISSUE_OPEN = Color.green.getRGB();
+    private static final int ISSUE_CLOSE = Color.red.getRGB();
+
+    /**
+     * A constant representing the date and time formatter used for formatting the creation date of
+     * an issue. The pattern "dd MMM, yyyy" represents the format "09 Oct, 2023".
+     */
+    static final DateTimeFormatter FORMATTER =
+            DateTimeFormatter.ofPattern("dd MMM, yyyy").withZone(ZoneOffset.UTC);
+    private final Predicate<String> hasGithubIssueReferenceEnabled;
+    private final Config config;
+
+    /**
+     * The repositories that are searched when looking for an issue.
+     */
+    private List<GHRepository> repositories;
+
+    public GitHubReference(Config config) {
+        this.config = config;
+        this.hasGithubIssueReferenceEnabled =
+                Pattern.compile(config.getGitHubReferencingEnabledChannelPattern())
+                    .asMatchPredicate();
+
+        acquireRepositories();
+    }
+
+    /**
+     * Acquires the list of repositories to use as a source for lookup.
+     */
+    private void acquireRepositories() {
+        try {
+            repositories = new ArrayList<>();
+
+            GitHub githubApi = GitHub.connectUsingOAuth(config.getGitHubApiKey());
+
+            for (long repoId : config.getGitHubRepositories()) {
+                repositories.add(githubApi.getRepositoryById(repoId));
+            }
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || !isAllowedChannelOrChildThread(event)) {
+            return;
+        }
```

Comment:
currently `#issueNumber` is only allowed in certain channels, but it's possible people might use em in other channels and then without any indication they won't know where to use this command. Imo we should respond with an ephermal response here something along the lines, "Hey, you used github reference feature in wrong channel. Please use `#allowedChannel`"

On a second note, why block this in other channels.
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/7289c210-e529-4299-917b-1697403bc967)
These are allowed channels currently.
Thoughts ?